### PR TITLE
Clean up the CSI KMS secret when the storage cluster is deleted

### PR DESCRIPTION
**Pull request by RishabhKodes**

### CODE REVIEW

File: controllers/storagecluster/kms_resources.go [hunk 1]
```diff
@@ -21,6 +21,8 @@ const (
 	CSIKMSConfigMapName = "csi-kms-connection-details"
 	// KMSTokenSecretName is the name of the secret which has KMS token details
 	KMSTokenSecretName = "ocs-kms-token"
+	// CephKMSTokenSecretName is the name of the ceph secret which has KMS token details
+	CephKMSTokenSecretName = "ceph-csi-kms"
```
Comment (GowthamShanmugam) [inline controllers/storagecluster/kms_resources.go:25]:
```suggestion
	CephKMSTokenSecretName = "ceph-csi-kms-token"
```

File: controllers/storagecluster/kms_resources.go [hunk 2]
```diff
@@ -21,6 +21,8 @@ const (
 	CSIKMSConfigMapName = "csi-kms-connection-details"
 	// KMSTokenSecretName is the name of the secret which has KMS token details
 	KMSTokenSecretName = "ocs-kms-token"
+	// CephKMSTokenSecretName is the name of the ceph secret which has KMS token details
```
Comment (GowthamShanmugam) [inline controllers/storagecluster/kms_resources.go:24]:
```suggestion
	// CSIKMSTokenSecretName is the name of the secret which has the  KMS token details of encrypted storage class
```

File: controllers/storagecluster/kms_resources.go [hunk 3]
```diff
@@ -21,6 +21,8 @@ const (
...
+	CephKMSTokenSecretName = "ceph-csi-kms-token"
```
Comment (GowthamShanmugam) [inline controllers/storagecluster/kms_resources.go:25]:
```suggestion
	CSIKMSTokenSecretName = "ceph-csi-kms-token"
```

File: controllers/storagecluster/kms_resources.go [hunk 4]
```diff
@@ -66,10 +68,17 @@ func deleteKMSResources(r *StorageClusterReconciler, sc *ocsv1.StorageCluster) e
 		return getKMSSecretToken(sc, client)
 	}
 
+	getCephKMSSecretTokenAsRuntimeObject := func(
```
Comment (GowthamShanmugam) [inline controllers/storagecluster/kms_resources.go:71]:
```suggestion
	getCSIKMSSecretTokenAsRuntimeObject := func(
```

File: controllers/storagecluster/kms_resources.go [hunk 5]
```diff
@@ -66,10 +68,17 @@ func deleteKMSResources(r *StorageClusterReconciler, sc *ocsv1.StorageCluster) e
...
+		sc *ocsv1.StorageCluster,
+		client client.Client) (client.Object, error) {
+		return getCephKMSSecretToken(sc, client)
```
Comment (GowthamShanmugam) [inline controllers/storagecluster/kms_resources.go:74]:
```suggestion
		return getCSIKMSSecretToken(sc, client)
```

File: controllers/storagecluster/kms_resources.go [hunk 6]
```diff
@@ -155,6 +164,23 @@ func getKMSSecretToken(instance *ocsv1.StorageCluster, client client.Client) (*c
 	return kmsSecretToken, err
 }
 
+// getCephKMSSecretToken function try to return the Ceph KMS Secret Token
```
Comment (GowthamShanmugam) [inline controllers/storagecluster/kms_resources.go:167]:
```suggestion
// getCSIKMSSecretToken function try to return the KMS Secret Token of encrypted storageClass
```

File: controllers/storagecluster/kms_resources.go [hunk 7]
```diff
@@ -155,6 +164,23 @@ func getKMSSecretToken(instance *ocsv1.StorageCluster, client client.Client) (*c
...
+func getCephKMSSecretToken(instance *ocsv1.StorageCluster, client client.Client) (*corev1.Secret, error) {
+	// if 'KMS' is not enabled, nothing to fetch
+	if !instance.Spec.Encryption.KeyManagementService.Enable {
+		return nil, nil
+	}
+	cephkmsSecretToken := &corev1.Secret{}
```
Comment (GowthamShanmugam) [inline controllers/storagecluster/kms_resources.go:173]:
```suggestion
	csiKMSSecretToken := &corev1.Secret{}
```

**Comment by GowthamShanmugam** [comment]
Change PR title "Clean up the CSI KMS secret when the storage cluster is deleted"

**Review by GowthamShanmugam — APPROVED** [review]

**Review by umangachapagain — COMMENTED** [review]
lgtm
Squash and sign the commits.

**Review by umangachapagain — APPROVED** [review]

**Pull request merged**